benchmarking/repos/hg.py

```diff
@@ -55,13 +55,14 @@ def getCommitTime(self, commit):
         end = t.index('<END>')
         return int(float(t[start:end]))
 
-    def getNextCommitHash(self, commit):
-        # always get the top-of-trunk commit
-        c = self._run('log', '--template', '<START>{node}<END>',
-                      '-r', 'tip')
-        start = c.index('<START>') + len('<START>')
-        end = c.index('<END>')
-        return c[start:end]
+    def getNextCommitHash(self, commit, step):
+        # always get the top-of-trunk commit if there is no more commits
+        command_str = f'descendants({commit}, {step})'
+        commits_str = self._run('log', '--template', '{node}:', '-r', command_str)
```

Comment:
You may want to do something like:

output = self._run('log', '--template', '<START>{node}<END>', '-r', commit)

Sometimes when you run the hg command, hg generates some warnings. Those warnings may mess up the string parsing. Need some guard to prevent that.